### PR TITLE
Refactored all the logging stuff into canteven-log.

### PR DESCRIPTION
**Pull request by taphu**
See commits for comments.

### CODE REVIEW

File: src/Canteven/Config.hs [hunk 1]
```diff
@@ -139,7 +49,7 @@ parseOptions = do
   case getOpt Permute options args of
     ([], [], []) ->
                  return "config.yml"
-    ([Config config], [], []) ->
+    ([Config config], _, []) ->
```
Comment (potomak) [inline src/Canteven/Config.hs]: What does this change means? Is it associated to the refactoring?

Comment (taphu) [inline src/Canteven/Config.hs]: rats, I forgot to mention this. I had intended to break this out into a different commit. Sorry. This allows additional command line arguments to be passed without causing canteven-config to crash the program.

Comment (potomak) [inline src/Canteven/Config.hs]: Ok, got it, thanks.
